addons/sys_intercom/fnc_configIntercom.sqf

```diff
@@ -0,0 +1,107 @@
+/*
+ * Author: ACRE2Team
+ * Configures the intercom system of a vehicle.
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorTarget] call acre_sys_intercom_fnc_configIntercom
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vehicle"];
+
+private _classname = typeOf _vehicle;
+private _intercoms = configFile >> "CfgVehicles" >> _classname >> "AcreIntercoms";
+private _intercomNames = [];
+private _intercomDisplayNames = [];
+private _intercomPositions = [];
+private _intercomExceptions = [];
+private _intercomLimitedPositions = [];
+private _numLimitedPositions = [];
+private _intercomConnectByDefault = [];
+private _unitsIntercom = [];
+
+for "_i" from 0 to ((count _intercoms) - 1) do {
```

Comment:
Use `configProperties` or `configClasses.`

Reply:
I do not understand why, I am only using the counters, not the values of a particular array

Reply:
This should be waaay faster, `for-from-to` is among the slowest of loops. You also get `_x` preassigned so don't need to do L32.
```sqf
{
    // _x is current config
} forEach (configProperties [configFile >> "CfgVehicles" >> _classname >> "AcreIntercoms", "isClass _x", true]);
```
And you are definitely using values, what's all those `getText/Array/Number` then. 😆